lib/rules/no-unassigned-vars.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * @fileoverview Rule to flag variables that are never assigned
+ * @author Jacob Bandes-Storch <https://github.com/jtbandes>
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('../types').Rule.RuleModule} */
+module.exports = {
+	meta: {
+		type: "problem",
```

Comment:
```suggestion
		type: "problem",
		dialects: ["typescript", "javascript"],
		language: "javascript",
```

If this rule is supporting TypeScript syntax then we must update the meta data as well?